src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/PEHeaders.cs

```diff
@@ -301,7 +314,12 @@ private ImmutableArray<SectionHeader> ReadSectionHeaders(ref PEBinaryReader read
 
             for (int i = 0; i < numberOfSections; i++)
             {
-                builder.Add(new SectionHeader(ref reader));
+                var sectionHeader = new SectionHeader(ref reader);
+                if (isCoffOnly && sectionHeader.VirtualSize != 0)
```

Comment:
https://github.com/dotnet/runtime/issues/112830#issuecomment-2677632672 said that `VirtualAddress` is the field that must be zero in object files, but [according to documentation](https://learn.microsoft.com/en-us/windows/win32/debug/pe-format#:~:text=This%20field%20is%20valid%20only%20for%20executable%20images%20and%20should%20be%20set%20to%20zero%20for%20object%20files.) it's `VirtualSize`.

Reply:
That sounds right. For `VirtualAddress` the doc says "For object files, this field is the address of the first byte before relocation is applied; for simplicity, compilers should set this to zero.". So yeah, we probably don't want to enforce zero for object files. Even for executable files this may be zero for some sections like `.debug`.